lib/internal/webstreams/adapters.js

```diff
@@ -491,7 +493,9 @@ function newReadableStreamFromStreamReadable(streamReadable, options = kEmptyObj
     pull() { streamReadable.resume(); },
 
     cancel(reason) {
-      destroy(streamReadable, reason);
+      const error = reason || new ERR_STREAM_PREMATURE_CLOSE();
+      onFinished(error);
+      destroy(streamReadable, error);
```

Comment:
On a second though, I feel like this is a breaking change and we probably don't want it for situations where readable stream is already finished because it might emit an extra `'error'`